cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/service/MetadataAdmin.java

```diff
@@ -80,4 +81,15 @@
    * @throws ApplicationNotFoundException if the application is not found
    */
   void removeTags(Id.NamespacedId appId, String ... tags) throws NotFoundException;
+
+  /**
+   * Execute search for metadata for particular type of CDAP object.
+   *
+   * @param searchQuery The query need to be executed for the search.
+   * @param type The particular type of CDAP object that the metadata need to be searched.
+   *
+   * @return a {@link Iterable} records for metadata search.
+   * @throws NotFoundException if there is not record found for particular query text.
+   */
+  Iterable<MetadataSearchResultRecord> searchMetadataOnType(String searchQuery, String type) throws NotFoundException;
```

Comment:
I think we need to have another method to search metadata across all target types.